### ISSUE TITLE
Seeking (via arrow keys / seekbar) while song is fading out keeps the song playing

### ISSUE THREAD

**Issue by PaperShreds** [fooyin/fooyin#215]
**Describe the bug**
When fading on pause is activated, pressing pause and then using the seekbar / arrow keys to navigate in the song will keep the song playing but on the volume that it had faded to before seeking.

**To Reproduce**
1 - Turn on fade on pause/stop (to reproduce more easily, use a long fade out time, like 1000ms)
2 - Choose a song and start to play it
3 - Press pause (either by a shortcut or the pause button)
4 - While the fade out is happening, use arrow keys or seekbar to seek forwards/backwards
5 - The song will keep playing even though the play/pause button suggests that it is paused, also the volume will be on the level that it faded to before seeking, pressing the play button will keep the song playing but on the louder volume again

**Expected behaviour**
The song should fade out, even when changing playback position while fading. It should keep fading out on the song position that it was paused on (example: if I press pause at 0:20, then go to 0:42 while it is fading out, the fade should sound like it is fading from 0:20 even tho the seekbar was set to 0:42 during the fade. Pressing play again should start the song at 0:42 then.) 

**Screenshots**
 - - -

**System:**
 - OS: EndeavourOS Linux x86_64, Kernel 6.10.3-arch1-2
 - fooyin version: 0.5.3

**Additional context**

**Comment by PaperShreds** [comment]
I might have an idea where this bug comes from: when using the seekbar, the song starts to play again. 

Also, when pressing play the song plays at a loud volume for a few milliseconds, then proceeds to fade in after that.

**Comment by ludouzi** [comment]
Are you using the waveform seekbar? I still need to push a fix related to that.

I'll look into the second.

**Comment by PaperShreds** [comment]
Yes, I am indeed using the waveform seekbar; I just noticed that the normal seekbar works as intended!

**Comment by PaperShreds** [comment]
Okay so I just rebuilt fooyin: the fade-out works with the waveform seekbar now, it also stays paused when seeking.
The fade-in bug still persists though.

**Comment by ludouzi** [comment]
I can reproduce the fade-in issue on some outputs so I'll reopen this. There's also problems when switching output/device when paused which needs looking into.

**Comment by PaperShreds** [comment]
Found another bug that relates to this one:
1 - Pause song with fade
2 - While fading, use seek bar (doesn't matter if waveform or normal)
3 - Keep the song paused, now select different song
4 - The next song will now start playing at the position that the previous song was seeked to and not from the beginning

**Comment by ludouzi** [comment]
Closing this as it appears to be resolved.
